### PR TITLE
Added commonName extension method to X500Name and helper class for x5…

### PR DESCRIPTION
**Pull request by shamsasari**
…09 cert factories

These changes will also be useful for the doorman.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/crypto/X509Utilities.kt [hunk 1]
```diff
@@ -609,3 +603,11 @@ object X509Utilities {
         return keyStore
     }
 }
+
+val X500Name.commonName: String get() = getRDNs(BCStyle.CN).first().first.value.toString()
+
+class CertificateStream(val input: InputStream) {
+    private val certificateFactory = CertificateFactory.getInstance("X.509", BouncyCastleProvider.PROVIDER_NAME)
```
Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/crypto/X509Utilities.kt]: Is it necessary to use BC here or is the X.509 support in the JDK sufficient?

Comment (mikehearn) [inline core/src/main/kotlin/net/corda/core/crypto/X509Utilities.kt]: Maybe just stick a TODO to revisit this in cleanup week.

File: node/src/main/kotlin/net/corda/node/utilities/certsigning/HTTPCertificateSigningService.kt [hunk 2]
```diff
@@ -49,7 +49,7 @@ class HTTPCertificateSigningService(val server: URL) : CertificateSigningService
         conn.outputStream.write(request.encoded)
 
         return when (conn.responseCode) {
-            HttpURLConnection.HTTP_OK -> IOUtils.toString(conn.inputStream)
+            HTTP_OK -> IOUtils.toString(conn.inputStream)
             HttpURLConnection.HTTP_FORBIDDEN -> throw IOException("Client version $clientVersion is forbidden from accessing permissioning server, please upgrade to newer version.")
```
Comment (patrickkuo) [inline node/src/main/kotlin/net/corda/node/utilities/certsigning/HTTPCertificateSigningService.kt]: can remove this as well? `HttpURLConnection `

**Comment by patrickkuo** [comment]
Looks like some problem with CI? Can we make sure this build please?

**Comment by shamsasari** [comment]
Comment addressed.

**Pull request merged**